working/macros/feature-specification.md

```diff
@@ -304,20 +304,109 @@ class, allowing you to build up your code fragments however you like.
 
 ## Scoping
 
-**TODO**: Fill in this section with more detail and a real proposal.
+### Resolved identifiers
 
 Macros will likely want to introduce references to identifiers that are not in
 the scope of the library in which they are running, but are in the scope of the
-macro itself.
+macro itself, or possibly even references which are not in scope of either the
+macro itself or the library where it is applied.
 
-We will want some way of providing an affordance to emit a reference to
-something from the macro scope, but in the code generated for the original
-library.
+Even if an identifier is expected to be in scope of the library in which the
+macro is running (lets say its exported by the macro library), that identifier
```

Comment:
"in which the macro is running" is vague since "running" is a dynamic and not textual concept. How about: 

> Even if an identifier is in scope where the macro is applied...